daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/AuthIT.scala

```diff
@@ -1,74 +1,77 @@
 // Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package com.daml.lf.engine.script.test
+package com.daml.lf.engine.script
+package test
 
+import com.daml.ledger.api.domain
 import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
-import com.daml.lf.data.ImmArray
-import com.daml.lf.data.Ref._
+import com.daml.lf.data.{ImmArray, Ref}
 import com.daml.lf.engine.script.ledgerinteraction.ScriptTimeMode
 import com.daml.lf.value.Value
-import com.daml.lf.speedy.SValue._
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AsyncWordSpec
 
 import java.io.File
 import scala.concurrent.Future
+import scala.util.{Failure, Success}
 
 final class AuthIT
     extends AsyncWordSpec
     with CantonFixture
     with Matchers
     with SuiteResourceManagementAroundAll {
+  import CantonFixture._
   import AbstractScriptTest._
 
-  // TODO: https://github.com/digital-asset/daml/issues/16539
-  //  The test is broken, the commented `transform` called in the for construct
+  lazy val secret = "secret"
+
+  override protected lazy val authSecret = Some(secret)
+  override protected lazy val darFiles: List[File] = List(stableDarPath)
+  override protected lazy val devMode: Boolean = false
+  override protected lazy val nParticipants: Int = 1
+  override protected lazy val timeMode = ScriptTimeMode.WallClock
+  override protected lazy val tlsEnable: Boolean = false
+
   "Daml Script against authorized ledger" can {
     "auth" should {
       "create and accept Proposal" in {
         for {
-          nonAuthenticatedClients <- participantClients()
-          grpcClient = nonAuthenticatedClients.default_participant.get
-          parties <- Future.sequence(
-            List.fill(2)(grpcClient.allocateParty("", ""))
-          )
-          value = Some(
-            Value.ValueRecord(
-              None,
-              ImmArray(
-                None -> Value.ValueParty(Party.assertFromString(parties.head)),
-                None -> Value.ValueParty(Party.assertFromString(parties.tail.head)),
-              ),
-            )
+          adminClient <- defaultLedgerClient(getToken(adminUserId))
+          userId = Ref.UserId.assertFromString(CantonFixture.freshName("user"))
```

Comment:
I think you meant to have this instead

```suggestion
          userId = Ref.UserId.assertFromString(CantonFixture.freshUserId())
```

Reply:
7019c4c743c1fa29702cfe1a8c94823d0babc7cb